modules/aws/volume-access/main.tf

```diff
@@ -0,0 +1,112 @@
+data "aws_caller_identity" "current" {}
+locals {
+  account_id                = data.aws_caller_identity.current.account_id
+  external_id               = (var.external_id != "" ? [{ test : "StringEquals", variable : "sts:ExternalId", values : [var.external_id] }] : [])
+  assume_conditions         = concat(local.external_id, local.source_identity, local.principal_check, local.vendor_federation)
+  support_assume_conditions = concat(local.external_id, local.source_identity)
+  source_identity           = (length(var.source_identities) > 0 ? [{ test : var.source_identity_test, variable : "sts:SourceIdentity", values : var.source_identities }] : [])
+  oidc_providers            = distinct(concat(var.oidc_providers, local.default_oidc_providers))
+  principal_check           = (length(var.streamnative_principal_ids) > 0 ? [{ test : "StringLike", variable : "aws:PrincipalArn", values : var.streamnative_principal_ids }] : [])
+  tag_set                   = merge({ Vendor = "StreamNative", Module = "StreamNative Volume", SNVersion = var.sn_policy_version }, var.tags)
+  vendor_federation         = (var.enforce_vendor_federation ? [{ test : "StringLike", variable : "aws:FederatedProvider", values : ["accounts.google.com"] }] : [])
+  # Add streamnative default eks oidc provider
+  default_oidc_providers = compact([
+
+  ])
+  conditions = [
+    for value in local.oidc_providers :
+    [
+      {
+        provider : "${value}",
+        test : "StringEquals",
+        variable : "${value}:aud",
+        values : ["sts.amazonaws.com"]
+      },
+      {
+        provider : "${value}",
+        test : "StringEquals",
+        variable : "${value}:sub",
+        values : [format("system:serviceaccount:%s:*", var.external_id)]
+      }
+    ]
+  ]
+}
+
+resource "aws_iam_openid_connect_provider" "streamnative_oidc_providers" {
```

Comment:
We should first check if the oidc provider exist in caller account or not, then only create the oidc provider if not exist.

Reply:
Fixed